nefertari_sqla/documents.py

```diff
@@ -24,10 +25,28 @@
 from .signals import ESMetaclass, on_bulk_delete
 from .fields import ListField, DictField, IntegerField
 from . import types
+from nefertari_sqla.utils import SingletonMeta
 
 log = logging.getLogger(__name__)
 
 
+class SessionHolder(metaclass=SingletonMeta):
+
+    def __init__(self):
+        # use pyramid_sqlaclhemy default session factory
+        self.session_factory = Session
+
+    def set_session_factory(self, factory):
+        self.session_factory = factory
+
+    def __call__(self):
+        return self.session_factory()
+
+    def restore_default(self):
+        from pyramid_sqlalchemy import Session
+        self.session_factory = Session
```

Comment:
If I remember correctly, you implemented a very similar class elsewhere (maybe in Celery tasks?)

Do you think you could use the same class everywhere?